### PR TITLE
strip comments of public keys before adding them to the ldap

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
cc @stdweird

File: lib/vsc/administration/ldapsync.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def sync_altered_accounts(self, last, dry_run=True):
                 logging.warning("Converting unicode to ascii for gecos resulting in %s", gecos)
             logging.debug('fetching public key')
 
-            public_keys = [str(x.pubkey) for x in self.client.get_public_keys(account.vsc_id)]
+            public_keys = process_public_keys([str(x.pubkey) for x in self.client.get_public_keys(account.vsc_id)])
```
Comment (stdweird) [inline lib/vsc/administration/ldapsync.py:124]: process one key at the time in the comprehension?

File: lib/vsc/administration/tools.py [hunk 2]
```diff
@@ -45,3 +46,21 @@ def create_stat_directory(path, permissions, uid, gid, posix, override_permissio
...
     return posix.create_stat_directory(path, permissions, uid, gid, override_permissions=override_permissions)
+
+
+def process_public_keys(public_keys):
+    """
+    Strip off the comment for a public key
+    @param public_keys list of ssh keys from the accountpage
+    @return list with public keys
+    """
+    ldap_public_keys = []
```
Comment (stdweird) [inline lib/vsc/administration/tools.py:57]: rename the variable, has nothing to do wih ldap

File: lib/vsc/administration/tools.py [hunk 3]
```diff
@@ -45,3 +46,21 @@ def create_stat_directory(path, permissions, uid, gid, posix, override_permissio
...
+    ssh_key = re.compile(r'^([^\s]+\s)?(?P<key>ssh-[^\s]+ [^\s]+)(\s.*)?$')
```
Comment (stdweird) [inline lib/vsc/administration/tools.py:58]: should be stored in a constant, use non-matching groups also?

File: lib/vsc/administration/tools.py [hunk 4]
```diff
@@ -45,3 +46,21 @@ def create_stat_directory(path, permissions, uid, gid, posix, override_permissio
...
+    for key in public_keys:
+        find_key = ssh_key.search(key)
+        if find_key:
+            ldap_public_keys.append(find_key.group('key'))
+        else:
```
Comment (stdweird) [inline lib/vsc/administration/tools.py:63]: log the else block, but what key would this be?

File: test/tools.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Tests for vsc.administration.tools
+
+@author Ward Poelmans (Vrije Universiteit Brussel)
+"""
+from vsc.install.testing import TestCase
+
+from vsc.administration.tools import process_public_keys
+
+
+test_pubkeys = [
```
Comment (stdweird) [inline test/tools.py:25]: add a test with some command/from directives

**Comment by stdweird** [comment]
not in favour, but if we choose for this, it's probably ok, and we need some changes in docs or AP to indictae we do this.

**Comment by itkovian** [comment]
I actually disagree with this.
- UA and KUL use the LDAP entries to sync keys to the system, users may want to see the comment to better ID which keys there are
- Stripping this in the LDAP makes no sense, if you do not want the comment, strip it in the AP when the keys are uploaded.

**Comment by wpoely86** [comment]
Yeah, not needed any more now we can do socket activation of sssd-ssh.

But we if go this route fully, the users will not see their keys any more on the system.

**Pull request closed without merging**